aws/cloudformation/serverless/serverless-blog-s3.yml

```diff
@@ -0,0 +1,294 @@
+                  - s3:GetObjectVersion
+                  - s3:GetBucketVersioning
+              - Resource: "*"
+                Effect: Allow
+                Action:
+                  - codebuild:StartBuild
+                  - codebuild:BatchGetBuilds
+                  - codedeploy:*
+                  - cloudformation:*
+
+
+#--[ CodeBuild Production ]--------------------------------------------------------
+  
+  CodeBuild:
+    Type: AWS::CodeBuild::Project
+    Properties:
+      Artifacts:
+        Type: "CODEPIPELINE"
+      Source:
+        Type: "CODEPIPELINE"
+      Environment:
+        ComputeType: "BUILD_GENERAL1_SMALL"
+        Image: "node:8-alpine"
+        Type: "LINUX_CONTAINER"
+        EnvironmentVariables:
+          - Name: AWS_DEFAULT_REGION
+            Value: !Ref AWS::Region
+          - Name: BUCKET_NAME
+            Value: !Ref BucketName
+          - Name: CACHE_CONTROL_MAX_AGE
+            Value: !Ref CacheControlMaxAge
+          - Name: BRANCH
+            Value: !Ref GitHubBranch
+          - Name: REPO
+            Value: !Ref GitHubRepo
+          - Name: USER
+            Value: !Ref GitHubUser
+      Name: !Ref GitHubRepo
+      ServiceRole: !GetAtt CodeBuildServiceRole.Arn
+
+#--[ CodePipeline ]-------------------------------------------------------------
+  CodePipeline:
+    Type: AWS::CodePipeline::Pipeline
+    Properties:
+      Name: !Ref GitHubRepo
+      RoleArn: !GetAtt CodePipelineServiceRole.Arn
+      ArtifactStore:
+        Type: S3
+        Location: !Ref BuildBucket
+      Stages:
+        - Name: "Source"
+          Actions:
+            - Name: "Download_Source"
+              ActionTypeId:
+                Category: Source
+                Owner: ThirdParty
+                Version: 1
+                Provider: GitHub
+              Configuration:
+                Owner: !Ref GitHubUser
+                Repo: !Ref GitHubRepo
+                Branch: !Ref GitHubBranch
+                OAuthToken: !Ref GitHubToken
+              OutputArtifacts:
+                - Name: source
+              RunOrder: 1
+        - Name: "Build"
+          Actions:
+            - Name: "Build_and_deploy"
+              ActionTypeId:
+                Category: Build
+                Owner: AWS
+                Version: 1
+                Provider: CodeBuild
+              Configuration:
+                ProjectName: !Ref CodeBuild
+              InputArtifacts:
+                - Name: source
+              OutputArtifacts:
+                - Name: build
+              RunOrder: 1
+
+#--[ Content delivery ]--------------------------------------------------
+  
+  CloudFront:
+    Type: 'AWS::CloudFront::Distribution'
+    Properties:
+      DistributionConfig:
+        Origins:
+        - DomainName: !Select [2, !Split ["/", !GetAtt S3WebsiteBucket.WebsiteURL]]
+          Id: S3Redirection
+          CustomOriginConfig:
+            HTTPPort: '80'
+            HTTPSPort: '443'
```

Comment:
Not mandatory :upside_down_face: 
HTTP and HTTPS are by default on those ports.